### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by speth**
**Changes proposed in this pull request**

- Update to build docs with Sphinx 5.1.1 as part of CI
- Remove obsolete Doxygen options that generated warnings with Doxygen 1.9.6
- Fix alphabetization of phase models in Sphinx docs
- Note version added for recent ThermoPhase models
- Use Doxylink to link to C++ classes instead of using direct HTML links to the dev docs
- Revise `ThermoPhase` documentation and eliminate the separate, outdated `thermoprops.dox` file
- Fix instances of `\log` that should be `\ln`, and always write `\log_{10}` when we mean the base 10 log.
- Add BibTeX citations for some transport-related classes

**If applicable, fill in the issue number this pull request is fixing**

Closes #Cantera/enhancements#175

**Checklist**

- [x] The pull request includes a clear description of this code change
- [x] Commit messages have short titles and reference relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] Style & formatting of contributed code follows [contributing guidelines](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)
- [x] The pull request is ready for review

### CODE REVIEW

File: doc/doxygen/thermoprops.dox
Comment (ischoegl) [inline doc/doxygen/thermoprops.dox:1]: Minor comment: an alternative to moving everything back to `*.h` files is to use markdown files (see examples in #1548).

**Review by ischoegl — DISMISSED** [review]
Thanks, @speth. I appreciate that additional documentation cleanup. Changes all look good to me.

File: doc/doxygen/cantera.bib [hunk 1]
```diff
@@ -131,13 +141,34 @@ @software{BOLOS
...
+@article{marrero1972,
+  title={Gaseous diffusion coefficients},
+  author={T.~R.~Marrero and E.~A.~Mason},
+  journal={Journal of Physical and Chemical Reference Data},
+  volume={1},
+  number={1},
+  pages={3--118},
+  doi = {10.1063/1.3253094},
+  url = {https://doi.org/10.1063/1.3253094},
+  year={1972}}
```
Comment (ischoegl) [inline doc/doxygen/cantera.bib]: minor indentation quibble.

File: doc/doxygen/thermoprops.dox
Comment (speth) [reply doc/doxygen/thermoprops.dox:1]: I did consider just converting this to a `.md` file, but in this instance I felt that so much of this was properly documentation for the `ThermoPhase` class, not just the `thermoprops` group, that it made more sense to move it all into `ThermoPhase.h`.

**Review by ischoegl — APPROVED** [review]

**Pull request merged**